src/router/storeSync/abstractParamConfig.class.js

```diff
@@ -201,4 +205,25 @@ export default class AbstractParamConfig {
             }
         })
     }
+
+    /**
+     * Triggers an action after the store has been populated with the query value. Returns a promise
+     *
+     * @returns {Promise<any>}
+     */
+    afterPopulateStore() {
+        return new Promise((resolve) => {
+            if (!this.afterSetValuesInStore) {
+                resolve()
+            }
+            const promiseAfterSetValuesInStore = this.afterSetValuesInStore()
+            if (promiseAfterSetValuesInStore) {
+                promiseAfterSetValuesInStore.then(() => {
+                    resolve()
+                })
+            } else {
+                resolve()
+            }
+        })
+    }
```

Comment:
you could maybe simplify the complexity by turning this into an `async` function

```
async afterPopulateStore() {
    if (!this.afterSetValuesInStore) {
        return
    }
    return await this.afterSetValuesInStore()
}
```
(might need some debug, I just tried transforming this on the fly without running it)